infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/VerifyingRemoteLeaderState.java

```diff
@@ -55,6 +55,9 @@ public LogReplicationRuntimeState processEvent(LogReplicationRuntimeEvent event)
                     return fsm.getStates().get(LogReplicationRuntimeStateType.WAITING_FOR_CONNECTIVITY);
                 }
                 return this;
+            case REMOTE_LEADER_LOSS:
+                this.worker.submit(this::verifyLeadership);
```

Comment:
this.worker.submit(this::verifyLeadership); can be removed right? by returning this? it will re-enter this state and execute that logic in the onEntry..

Reply:
ok